modules/remove_map/functions/RemoveMap/fn_RM_doRemoves.sqf

```diff
@@ -0,0 +1,35 @@
+// AUTHORS: TinfoilHate and StatusRed
+#include "script_component.hpp"
+
+params ["_keepMapClasses", "_keepCompassClasses"];
+
+_keepMapClasses append _keepCompassClasses;
+
+[{time > 1}, {
+	private _keepMapClasses = (_this select 0);
+	private _keepCompassClasses = (_this select 1);
+
+	{
+		_unit = _x;
+		_gear = (_unit getVariable [QGVAR(Loadout), ""]);
+		if (isNil _gear) then {
+			_gear = "NO VALUE FOUND";
+		};
+
+		if (local _unit) then {
+			private _inKeepMap = _gear in _keepMapClasses;
+			private _inKeepCompass = _gear in _keepMapClasses;
+
+			if (_inKeepMap) then {
+				if (_inKeepCompass) then {
+					_unit unlinkItem "ItemMap";
+				} else {
+					//Hooray, you're important!
+				};
+			} else {
+				_unit unlinkItem "ItemMap";
+				_unit unlinkItem "ItemCompass";
+			};
+		};
+	} forEach allUnits;
```

Comment:
Instead of using allUnits and just iterating through units, you can use the CBA initPost EH for the units which can do the processing locally and handle JIP/respawn.

Reply:
Can you give me an example of this please? I'm not sure how to use it.

Reply:
https://github.com/Global-Conflicts-ArmA/Olsen-Framework-Arma-3/blob/cb7b3f72478395034a3b1a6cba27502e54237525/core/description.hpp#L67

https://github.com/CBATeam/CBA_A3/wiki/Extended-Event-Handlers-(new)

Reply:
Done.